Update server to Python 3 + using yarn instead of npm

duraz0rz opened the pull request:
Python 2 is dead, EOL, and nothing should really run on it any more.  Wasn't too bad as there's a program that does the conversion for you, already.

I also removed the `original_simfiles` directory...nothing's using stuff from there, so might as well nuke it. You can always revert 
5a52b7e if you need it back for whatever reason.

Also switched to using the yarn package manager instead of npm because yarn is typically faster and uses less disk space. Usage is similar, too.

zachwalton commented:
Awesome, thanks for the contribution! I’ll test this out tomorrow to make sure it still builds + deploys with Heroku and will merge after.

zachwalton commented on README.md:
```diff
@@ -12,21 +12,27 @@ Pick a preferred read speed and select a song. You'll get a breakdown of BPM dur
-pip install -r requirements.txt -r dev_requirements.txt
-cd frontend/ && npm install && npm run build && cd -
-honcho start
+pyenv install (This'll install 3.7.5 if you don't have it already)
```
```suggestion
# Assuming you use Homebrew. Otherwise install pyenv / pyenv-virtualenv using your preferred package manager / method
brew install pyenv pyenv-virtualenv
pyenv install (This'll install 3.7.5 if you don't have it already)
```

duraz0rz replied:
I left the installation part out as `pyenv` is linked above in the requirements, so someone can just go there if they don't have it.

zachwalton replied:
Was trying to cover `pyenv-virtualenv` specifically since I ended up Googling install instructions after getting an error that it was missing.

zachwalton replied:
(not a big deal or blocker though)

zachwalton commented on README.md:
```diff
@@ -12,21 +12,27 @@ Pick a preferred read speed and select a song. You'll get a breakdown of BPM dur
 
+To serve the API + frontend, `./serve.sh`
+
 This will serve the API + frontend on `http://127.0.0.1:5000`. If you just want to work on the frontend, you may want to use npm to get the benefits of live reloading:
```
```suggestion
This will serve the API + frontend on `http://127.0.0.1:5000`. If you just want to work on the frontend, you may want to use yarn to get the benefits of live reloading:
```

zachwalton commented:
Actually this breaks some charts :(

Run `./serve.sh` locally and access: http://0.0.0.0:5000/#readSpeed=573&song=EGOISM%20440%20-%20U1%20High-Speed

```
08:54:31 web.1  | Traceback (most recent call last):
08:54:31 web.1  |   File "app.py", line 59, in api_v1_simfiles_name
08:54:31 web.1  |     "result": parsed.analyze(**query_params)
08:54:31 web.1  |   File "/Users/zachwalton/git/truebpm/simfile/parser.py", line 196, in analyze
08:54:31 web.1  |     num_measures, durations = self.bpm_durations(style, difficulty)
08:54:31 web.1  |   File "/Users/zachwalton/git/truebpm/simfile/parser.py", line 156, in bpm_durations
08:54:31 web.1  |     for measure, bpm in self.BPMS.items():
08:54:31 web.1  | AttributeError: 'SMParser' object has no attribute 'BPMS'
```

Reverting to old build for now

duraz0rz commented:
Ok, I think I addressed your feedback. I fixed the Egoism issue by breaking on a line that starts with `#NOTES` in `parse_section_header` instead of breaking on a comment that doesn't have `song id`. Added bonus is that all files will stop earlier, not just the ones with that `//----- song ID:` section header (5.1.1, for example).

zachwalton commented:
Looks good! Thanks again 👍
